### PR TITLE
Add Consumer Guide

### LINKED ISSUE #187: ❓👂 Question/Feedback - How to consume this repository?
## Question/Feedback

I have some general questions of best practice around ingesting and extending the output of this module. 

Fundamentally: 

* What are the options for consuming the outputs of this repository?
* What are the pros/cons for each?
* Are there product roadmap considerations? e.g. Bicep public module registry

Although generally implementation and decisions are down to the consumer of the module repos I haven't found it clear what the options are available which makes this harder to use.

## Possible Answers/Solutions?

* I could fork this repository (not marked as template though), then add my pipelines and extend the functionality.
* I have seen that there is documentation for private modules - so I could create an ACR publish these modules and consume in my own repository

Perhaps there are other options available as well that I am not considering?

Certainly there is a mix of approach in this repository with the majority of the examples and pipelines referencing the local modules and infact I have done both of the options that I suggested above with differing success.

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

This PR adds a new doc for the "Consumer Guide" inspired by issue #187 

## This PR fixes/adds/changes/removes

1. Fixes #187 
2. Adds missing Deployment Flow navigation from all menus following #152 

### Breaking Changes

None

## Testing Evidence

Docs only, linting will suffice.

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [ADO items](https://dev.azure.com/unifiedactiontracker/Solution%20Engineering/_backlogs/backlog/Azure%20Landing%20Zone%20Bicep/Backlog%20Bugs%20Feedback)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [ ] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

File: docs/wiki/ConsumerGuide.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## How to Consume `ALZ-Bicep`
+<!-- markdownlint-restore -->
+
+## Background
+
+> This guidance supports the [Deployment Flow](https://github.com/Azure/ALZ-Bicep/wiki/DeploymentFlow) guidance, it is not a replacement
+
+The `ALZ-Bicep` repository (this repository) has been created to help customers and partners to deploy and deliver the [Azure Landing Zones (ALZ) conceptual architecture](https://aka.ms/alz#azure-landing-zone-conceptual-architecture) into an Azure AD Tenant utilizing [Bicep](https://aka.ms/bicep) as the Infrastructure-as-Code (IaC) tooling and language.
+
+The style in which the Bicep modules have been authored in this repo are aimed at consumers of all skill levels. This is in an effort to make the modules as accessible as possible; especially for those that are newer to the world of IaC and/or Bicep. This is to help consumers accelerate their journey to Azure as ALZ is likely to be one of the key first pieces of Azure a customer or partner will deploy and configure to establish the platform guardrails, connectivity and security elements to enable the future success of workloads within Azure.
```
Comment (jfaurskov) [inline docs/wiki/ConsumerGuide.md:11]: Last sentence is a bit knotty, could perhaps be simplified?

File: docs/wiki/ConsumerGuide.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - [Azure Resource Manager template specs in Bicep](https://docs.microsoft.com/azure/azure-resource-manager/bicep/template-specs)
+- Use and reference the modules directly from the Microsoft Public Bicep Registry - ***Coming Soon (awaiting feature release in Bicep)***
+
+The option to use will be different per consumer based on their experience and skill levels with the various pieces of technology and their features.
+
+> Consumers can also use the above listed methods together with CI/CD pipelines to deploy the modules. See [Azure Landing Zones Bicep - Pipelines](https://github.com/Azure/ALZ-Bicep/wiki/PipelinesOverview) for more information and guidance
+
+## Customizing the `ALZ-Bicep` Modules
+
+Whatever way you may choose to consume the modules we do do expect, and want, customers and partners to customize the modules to suit their needs and requirements for their design in their local copies of the modules.
```
Comment (jfaurskov) [inline docs/wiki/ConsumerGuide.md:39]: do do

File: docs/wiki/ConsumerGuide.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+> **IMPORTANT:** If you believe the changes you have made should be more easily available to be customized by a parameter etc. in the module, then please raise an [issue](https://github.com/Azure/ALZ-Bicep/issues) for a 'Feature Request' on the repository 👍
+> 
+> If you wish to, also feel free to submit a pull request relating to the issue which we can review and work with you to potentially implement the suggestion/feature request 👍
+<!-- markdownlint-restore -->
+
+## Considerations when Customizing the `ALZ-Bicep` Module
+
+Whilst customizing the modules on the whole is straightforward and simple, there are some key things to consider depending on which modules you customize.
+
+> See something missing from here? Please raise an [issue](https://github.com/Azure/ALZ-Bicep/issues) on the repo and let us know and feel free to contribute a pull request also 👍
```
Comment (jfaurskov) [inline docs/wiki/ConsumerGuide.md:55]: Not sure if this is a typo or related to the above sentence.

**Review by jfaurskov — CHANGES_REQUESTED** [review]

File: docs/wiki/ConsumerGuide.md (hunk 3 above)
Comment (jtracey93) [reply docs/wiki/ConsumerGuide.md]: Nope this is valid here as want to know if customers think anything else should be added in this section

**Pull request merged**